### DISCUSSION TITLE
Device transport responses

### DISCUSSION THREAD

**Discussion by eloycoto** [RedHatInsights/yggdrasil#50] [Ideas]
Hi, 

Using MQTT, responses are not expected, but when using HTTP, maybe one device wants to have the response result, which can be any kind of information needed by the device. In my case, a signed certificate. 

Could we open a discussion about how to implement this? I was thinking to have an event interface per worker, but maybe it's an overhead. 

Related to: https://github.com/RedHatInsights/yggdrasil/pull/39
Sample POC using events: https://github.com/jakub-dzon/yggdrasil/pull/12

**Comment by subpop** [comment]
I _just_ finished writing an initial topic about the RPC protocol. See #51.
